lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -939,9 +1013,43 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
 
 LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
                                      CmpIOp op) const {
+  auto isPipeLibOp = [](Value val) -> bool {
+    if (Operation *defOp = val.getDefiningOp()) {
+      return isa<calyx::MultPipeLibOp, calyx::DivUPipeLibOp,
+                 calyx::DivSPipeLibOp, calyx::RemUPipeLibOp,
+                 calyx::RemSPipeLibOp>(defOp);
+    }
+    return false;
+  };
+
   switch (op.getPredicate()) {
-  case CmpIPredicate::eq:
+  case CmpIPredicate::eq: {
```

Comment:
Isn't this the case for any conditional predicate used in a Calyx `if`? Just trying to make sure I understand correctly what's going on here. If this is true, don't feel pressured to make those changes now but you should add some assertions so that there is some sort of failure.

Reply:
You are right, any comparison operator can be a condition check.
I'm only adding `eq` for now to use it as a prototype since some approaches need to be discussed (like the `PipeLoweringState`). Once we reach agreement, I will add support for all `CmpIPredicate`, which will just be cloning the same code from `eq`.